backend/reservations/views.py

```diff
@@ -1,3 +1,286 @@
+                                   key, algorithm="HS256")
+                return Response(str(json.dumps({"jwt": resp})), status=status.HTTP_201_CREATED)
+        return Response("Invalid credentials", status=status.HTTP_400_BAD_REQUEST)
+
+
+def phrasing():
+    words = None
+    with open("./wordylyst.md", "r") as f:
+        words = [word for word in f.read().splitlines()]
+    word = words[random.randint(0, 999)].capitalize()+words[random.randint(0, 999)].capitalize()
+    rand = random.randint(1,3)
+    if(rand==1):
+        word = word+str(random.randint(0,99))
+    elif(rand==2):
+        word = word+str(random.randint(0,99))+words[random.randint(0,999)].capitalize()
+    else:
+        word = word+words[random.randint(0,999)].capitalize()
+    return word
+
+
+@api_view(['POST'])
+def swap_gen(request):
+    if request.method == 'POST':
+        data = request.data
+        try:
+            jwt_data=data["jwt"]
+            room_num=data["number"]
+        except KeyError as e:
+            return Response("missing fields", status=status.HTTP_400_BAD_REQUEST)
+        email = validate_jwt(jwt_data)
+        if (email is None):
+            return Response("Invalid jwt", status=status.HTTP_400_BAD_REQUEST)
+        try:
+            guest_instances = Guest.objects.filter(email=email)
+            guest_id = guest_instances[0].id
+        except IndexError as e:
+            return Response("No guest found", status=status.HTTP_400_BAD_REQUEST)
+        rooms = Room.objects.filter(guest=guest_id)
+        phrase = phrasing()
+        for room in rooms:
+            if(str(room_num) == str(room.number) ):
+                room.swap_code=phrase
+                room.swap_time=datetime.datetime.utcnow()
+                room.save()
+
+        response = json.dumps({"swap_phrase": phrase}, indent=2)
+        return Response(response)
+
+
+@api_view(['POST'])
+def swap_it_up(request):
+    if request.method == 'POST':
+        data = request.data
+        try:
+            jwt_data=data["jwt"]
+            room_num=data["number"]
+            swap_req=data["swap_code"]
+        except KeyError as e:
+            return Response("missing fields", status=status.HTTP_400_BAD_REQUEST)
+        email = validate_jwt(jwt_data)
+        if (email is None):
+            return Response("Invalid jwt", status=status.HTTP_400_BAD_REQUEST)
+        try:
+            guest_instances = Guest.objects.filter(email=email)
+            guest_id = guest_instances[0].id
+        except IndexError as e:
+            return Response("No guest found", status=status.HTTP_400_BAD_REQUEST)
+
+        rooms_mine = Room.objects.filter(guest=guest_id)
+        for room in rooms_mine:
+            if(str(room.number)==str(room_num)):
+                swap_room_mine = room
+            else:
+                return Response("Lies", status=status.HTTP_400_BAD_REQUEST)
+    
+
+        rooms_swap_match = Room.objects.filter(swap_code=swap_req)
```

Comment:
@clark this is where I grab the swapping room when a code is entered. I was also initially concerned about explicitly specifying the room number of room to be swapped but it seems like probability of getting the wrong room would be a function of collisions with the swap_phrase. Which I guess is the question right(?) Do we think there is sufficiently low probability of key collisions that we know we will only ever have one room number associated with one swap_code(?)

And it seems like trusting input from a user is always questionable. 

But let me know what u think, if we should still specify the room num to be swapped!